Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp

```diff
@@ -114,8 +162,16 @@ class IconCache {
             close(fd);
             return status == CAIRO_STATUS_SUCCESS ? filename.get() : CString();
 #elif USE(SKIA)
-            // FIXME: Add Skia implementation
-            return CString();
+            auto stream = fdWStream(fd);
+            SkPixmap pixmap;
+            if (!surface->peekPixels(&pixmap) || !SkPngEncoder::Encode(&stream, pixmap, { })) {
+                g_warning("Failed to encode notification icon to PNG");
+                close(fd);
```

Comment:
Give `fdWStream` a virtual destructor and close the fd there. i.e. let the stream object take ownership of the fd.